pkgs/development/compilers/rust/bootstrap.nix

```diff
@@ -16,19 +16,19 @@ let
   };
 
   platform =
-    if stdenv.system == "i686-linux"
+    if stdenv.hostPlatform.system == "i686-linux"
     then "i686-unknown-linux-gnu"
-    else if stdenv.system == "x86_64-linux"
+    else if stdenv.hostPlatform.system == "x86_64-linux"
     then "x86_64-unknown-linux-gnu"
-    else if stdenv.system == "armv7l-linux"
+    else if stdenv.hostPlatform.system == "armv7l-linux"
     then "armv7-unknown-linux-gnueabihf"
-    else if stdenv.system == "aarch64-linux"
+    else if stdenv.hostPlatform.system == "aarch64-linux"
     then "aarch64-unknown-linux-gnu"
-    else if stdenv.system == "i686-darwin"
+    else if stdenv.hostPlatform.system == "i686-darwin"
     then "i686-apple-darwin"
-    else if stdenv.system == "x86_64-darwin"
+    else if stdenv.hostPlatform.system == "x86_64-darwin"
     then "x86_64-apple-darwin"
-    else throw "missing bootstrap url for platform ${stdenv.system}";
+    else throw "missing bootstrap url for platform ${stdenv.hostPlatform.system}";
```

Comment:
@Ericson2314 Ah I think I know why I was confused about it now, this is an example of a binary tarball, there's no build steps except patching binaries. This and any other cases like it should be using targetPlatform, right?

Reply:
Nope, host platform. @Dezeg is right that "target" sounds great but isn't what you want, and is therefore sort of a trap.

Reply:
just remember target platform is "bad", and then 99% of cases are handled by process of elimination.